archaius2-test/src/main/java/com/netflix/archaius/test/Archaius2TestConfig.java

```diff
@@ -0,0 +1,289 @@
+package com.netflix.archaius.test;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.commons.lang3.ClassUtils;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import com.netflix.archaius.api.Config;
+import com.netflix.archaius.api.ConfigListener;
+import com.netflix.archaius.api.Decoder;
+import com.netflix.archaius.api.StrInterpolator;
+import com.netflix.archaius.config.MapConfig;
+
+/***
+ * JUnit rule which builds an Archaius2 {@link Config} instance using
+ * annotations.
+ * 
+ * <pre>
+ *  {@literal @}Rule
+ *  public Archaius2TestConfig config = new Archaius2TestConfig();
+ *
+ *  {@literal @}Test 
+ *  {@literal @}TestPropertyOverride({"propName=propValue"})
+ *  public void testBasicPropertyResolution() {
+ *      assertEquals("propValue", config.getString("propName"));
+ *  }
+ * </pre>
+ * 
+ * See {@link TestPropertyOverride} for additional usage information.
+ */
+public class Archaius2TestConfig implements TestRule, Config {
+    private final ThreadLocal<Config> configHolder = new ThreadLocal<>();
+
+    public Statement apply(Statement base, Description description) {
+        return statement(base, description);
+    }
+
+    private Statement statement(final Statement base, final Description description) {
```

Comment:
The finality of arguments is specific to the method implementation.  It does not matter how the interface declares them.